pennylane/estimator/templates/subroutines.py

```diff
@@ -971,6 +971,189 @@ def resource_decomp(cls, base_cmpr_op, num_iter):
         return gate_counts
 
 
+class UnaryIterationQPE(ResourceOperator):
+    r"""Resource class for UnaryIterationQPE. This is a variant of the Quantum Phase Estimation algorithm
+    that uses the unary iteration technique described in `arXiv. 2011.03494 <https://arxiv.org/pdf/2011.03494>`_.
+    In this approach, the Select operation is implemented using unary iteration, which reduces
+    T and Toffoli gate counts at the cost of increasing the number of ancilla qubits required. For
+    its typical usage, see the explanation at `<https://pennylane.ai/compilation/unary-iteration>`_.
+
+    Note we also provide the option to provide a custom adjoint QFT implementation, which can be used to
+    further optimize the resource requirements. For example, one could use an approximate QFT implementation
+    to reduce the T-gate count further.
+
+    Args:
```

Comment:
You could also add a `.. seealso::` section where you link to the estimator QPE or the pennylane QPE operators. Or even the Unary iteration compilation page.